packages/dataviews/src/components/dataviews/index.tsx

```diff
@@ -49,6 +49,8 @@ type DataViewsProps< Item > = {
 	selection?: string[];
 	onChangeSelection?: ( items: string[] ) => void;
 	header?: ReactNode;
+	isShowingFilter?: boolean;
+	setIsShowingFilter?: React.Dispatch< React.SetStateAction< boolean > >;
```

Comment:
I would love if avoid these props please. I don't think we should be adding random props like that to the DataViews component.

Why do we need them?

Reply:
Hey @youknowriad, we added these props to reset the isShowingFilter state when the active view changes (e.g: from drafts, to scheduled). This allows us to control or reset the state from outside of the component. If these props aren't passed, the component will use its own state, but having them allows outside control when necessary.

Reply:
I see. I think it's not ideal in terms of Api for the DataViews component. Why control this state and not other states like dropdown states... I think this doesn't make for a great API in DataViews.

Two options from me:

 - Can we accept that this remains an internal state.
 - If it's about resetting the state, we can force remount with a separate `key` on DataViews if needed.

Reply:
> If it's about resetting the state, we can force remount with a separate key on DataViews if needed.

I have thought of this option but I think it will probably also have undesired effects. I can give it a try and see how it goes.